src/main/java/teammates/ui/webapi/action/GetFeedbackSessionsAction.java

```diff
@@ -20,12 +23,27 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
+        if (!(userInfo.isStudent || userInfo.isInstructor)) {
+            throw new UnauthorizedAccessException("Student or instructor privilege is required to access this resource.");
+        }
+
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
-        if (courseId == null) {
-            gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
-        } else {
-            CourseAttributes courseAttributes = logic.getCourse(courseId);
-            gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(courseId, userInfo.getId()), courseAttributes);
+        if (userInfo.isInstructor) {
+            if (courseId == null) {
+                gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
```

Comment:
Is this really necessary? Doesn't usesrInfo.isInstructor perform this check already?